watcher/src/metrics.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Watcher metrics comparing ledger height and block height
+
+use mc_common::HashMap;
+use mc_util_metrics::{IntGauge, OpMetrics};
+use url::Url;
+
+lazy_static::lazy_static! {
+    /// Create metric object for tracking watcher
+    pub static ref COLLECTOR: OpMetrics = OpMetrics::new_and_registered("watcher");
+}
+
+/// Watcher metrics tracker used to report metrics on watcher to Prometheus
+pub struct WatcherMetrics {
+    /// Number of blocks in the ledger
+    pub ledger_block_height: IntGauge,
```

Comment:
Optional/Equivalent: Rather than surfacing this gauge (`pub`), add a `set_block_height` method that updates the gauge

Reply:
Good suggestion